app/preact/src/server/framework-preset-preact.ts

```diff
@@ -9,3 +11,20 @@ export function babelDefault(config: TransformOptions) {
     ],
   };
 }
+
+export function webpackFinal(config: Configuration) {
+  return {
+    ...config,
+    resolve: {
+      ...config.resolve,
+      // Prefer top-level aliases
+      modules: [path.resolve('node_modules'), ...config.resolve.modules],
+      // Alias react imports to "preact/compat":
+      alias: {
+        ...config.resolve.alias,
+        react: require.resolve('preact/compat'),
+        'react-dom': require.resolve('preact/compat'),
```

Comment:
```suggestion
        react: path.dirname(require.resolve('preact/compat/package.json')),
        'react-dom': path.dirname(require.resolve('preact/compat/package.json')),
```
My bad, needs to get the entry path otherwise it will load from `main` instead of letting webpack load `module`

Reply:
same thing
it creates nice path
but the end result is that ui is stuck
It compiles well, shows storybook controls and then stuck with a loader...